plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -2071,6 +2078,11 @@ private Session dynamicFiltering(JoinDistributionType joinDistributionType, bool
                 .build();
     }
 
+    /**
+     * This method relies on global state of QueryTracker. It may fail because of QueryTracker.pruneExpiredQueries()
+     * You must ensure that query was issued and this method invoked in isolation -
+     * which guarantees that there is less other queries between query creation and obtaining query info than `query.max-history`
+     */
```

Comment:
Should you add `TODO` and mention https://github.com/trinodb/trino/issues/18499? I would do it.